### PR TITLE
Adds unified sheet import

### PR DESCRIPTION
**Pull request by storytellermahkasad**
### Summary
Adds a unified sheet importing that discovers the correct sheet type to parse with.

#### PR Type

- [ ] This PR is a code change that implements a feature request.
- [ ] This PR fixes an issue.
- [X] This PR adds a new feature that is not an open feature request.
- [ ] This PR is not a code change (e.g. documentation, README, ...)
#### Other
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] If code changes were made then they have been tested.
- [ ] I have updated the documentation to reflect the changes.

### CODE REVIEW

File: cogs5e/sheetManager.py [hunk 1]
```diff
@@ -502,6 +502,61 @@ async def _confirm_overwrite(self, ctx, _id):
...
+
+        __Sheet-specific Notes__
+        D&D Beyond:
+            Private sheets can be imported if you have linked your DDB and Discord accounts.  Otherwise, the sheet needs to be publicly shared.
+        Gsheet:
+            The sheet must be shared with Avrae for this to work.
+            Avrae's google account is `avrae-320@avrae-bot.iam.gserviceaccount.com`.
+
+        Dicecloud:
+            Share your character with `avrae` on Dicecloud (edit perms) for live updates.
```
Comment (zhudotexe) [inline cogs5e/sheetManager.py:526]:
```suggestion
            Share your character with `avrae` on Dicecloud (edit permissions) for live updates.
```

File: cogs5e/sheetManager.py [hunk 2]
```diff
@@ -502,6 +502,61 @@ async def _confirm_overwrite(self, ctx, _id):
...
+        """
+        loading = await ctx.send('Loading character data to Avrae...')
+        prefix = ''
```
Comment (zhudotexe) [inline cogs5e/sheetManager.py:529]: What is this variable doing?

File: cogs5e/sheetManager.py [hunk 3]
```diff
@@ -502,6 +502,61 @@ async def _confirm_overwrite(self, ctx, _id):
...
+        parser = None
+
+        url = await self._check_url(ctx, url) # check for < >
+        # Sheets in order: DDB, Dicecloud, Gsheet
+        beyond_url = DDB_URL_RE.match(url)
```
Comment (zhudotexe) [inline cogs5e/sheetManager.py:534]: so I know this is just copied from the respective existing import methods, but I think we can improve this, lol. I think the best way to do this is by parsing the url using [urllib](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) and checking the `netloc`

File: cogs5e/sheetManager.py (hunk 2 above)
Comment (storytellermahkasad) [reply cogs5e/sheetManager.py:529]: It gets assigned per sheet type and is used in the confirm overwrites L554 but could refactor the block to not use it.

Comment (zhudotexe) [reply cogs5e/sheetManager.py:529]: Oh, there it is, just didn't see it

File: cogs5e/sheetManager.py [hunk 4]
```diff
@@ -502,6 +502,61 @@ async def _confirm_overwrite(self, ctx, _id):
...
+        Gsheet:
+            The sheet must be shared with Avrae for this to work.
+            Avrae's google account is `avrae-320@avrae-bot.iam.gserviceaccount.com`.
+
+        Dicecloud:
+            Share your character with `avrae` on Dicecloud (edit permissions) for live updates.
+        """
+        loading = await ctx.send('Loading character data to Avrae...')
+        prefix = ''
+        parser = None
```
Comment (zhudotexe) [inline cogs5e/sheetManager.py:530]: These 3 lines can be removed (the first sends a duplicate message, and the latter two are guaranteed to be set in one of the if bodies)

Comment (zhudotexe) [reply cogs5e/sheetManager.py:530]: Although the message needs to be moved before `extract_gsheet_id_from_url` in the google case in this case

**Pull request merged**